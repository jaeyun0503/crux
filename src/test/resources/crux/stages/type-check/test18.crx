// test loop

fn main() {
  let a: int;
  a = 0;
  while (a < 3) {
    printInt(a);
    println();
    a = a + 1;
  }
}
