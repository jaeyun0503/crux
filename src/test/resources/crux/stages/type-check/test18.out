Crux Program has no type errors.

main() {
  int a;
  /*int*/ a = /*int*/ 0;
  while (/*bool*/ (/*int*/ a < /*int*/ 3)) {
    printInt(/*int*/ a);
    println();
    /*int*/ a = /*int*/ (/*int*/ a + /*int*/ 1);
  }
}

